src/utilities/assertValidName.js

```diff
@@ -22,11 +25,20 @@ export function assertValidName(
       `Must be named. Unexpected name: ${name}.`
     );
   }
-  if (!isIntrospection && name.slice(0, 2) === '__') {
-    throw new Error(
-      `Name "${name}" must not begin with "__", which is reserved by ` +
-      'GraphQL introspection.'
-    );
+  if (!isIntrospection && name.slice(0, 2) === '__' && !hasWarnedAboutDunder) {
+    hasWarnedAboutDunder = true;
+    /* eslint-disable no-console */
+    if (console && console.error) {
```

Comment:
Just being paranoid. I'd rather not have a cascading failure because someone's running graphql-js in some environment that either has no `console` or has `console` but doesn't have `console.error`.